powersimdata/design/generation/curtailment.py

```diff
@@ -2,31 +2,15 @@
 
 from powersimdata.scenario.scenario import Scenario
 
-default_pmin_dict = {
-    "coal": None,
-    "dfo": 0,
-    "geothermal": 0.95,
-    "hydro": None,
-    "ng": 0,
-    "nuclear": 0.95,
-    "other": 0,
-    "solar": 0,
-    "wind": 0,
-    "wind_offshore": 0,
-}
-profile_methods = {
-    "hydro": "get_hydro",
-    "solar": "get_solar",
-    "wind": "get_wind",
-    "wind_offshore": "get_wind",
-}
-
 
 def temporal_curtailment(
-    scenario, pmin_by_type=None, pmin_by_id=None, curtailable={"solar", "wind"}
+    scenario,
+    pmin_by_type={"hydro": None},
+    pmin_by_id=None,
+    curtailable={"solar", "wind"},
```

Comment:
Not sure whether we would like to keep mutable defaults here.

Reply:
I was trying to keep the default behavior of the function since we added `hydro` to the list of curtailable resources.

Reply:
Yeah, I see where you come from. Just wonder whether we should move mutable defaults into `if` logic in the function as we did it elsewhere, such as `pmin_by_type = pmin_by_type if pmin_by_type else {"hydro": None}`.

Reply:
Done